### PR TITLE
chore(tests): enable lint, make it green.

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/@angular/compiler/test/css_visitor_spec.ts [hunk 1]
```diff
@@ -22,6 +22,9 @@ function _assertTokens(tokens: CssToken[], valuesArr: string[]): void {
 class MyVisitor implements CssAstVisitor {
   captures: {[key: string]: any[]} = {};
 
+  /**
+   * @internal
+   */
   _capture(method: string, ast: CssAst, context: any) {
```
Comment (alexeagle) [inline modules/@angular/compiler/test/css_visitor_spec.ts]: what's the point of visibility rules in spec files?

File: modules/@angular/compiler/test/test_component_builder_spec.ts [hunk 2]
```diff
@@ -328,291 +326,289 @@ export function main() {
...
+      it('should signal through whenStable when the fixture is stable (autoDetectChanges)',
+         inject(
+             [TestComponentBuilder, AsyncTestCompleter],
+             (tcb: TestComponentBuilder, async: AsyncTestCompleter) => {
 
-                 tcb.createAsync(AsyncComp).then((componentFixture) => {
-                   componentFixture.detectChanges();
-                   expect(componentFixture.nativeElement).toHaveText('1');
+               tcb.createAsync(AsyncComp).then((componentFixture) => {
+                 componentFixture.autoDetectChanges();
```
Comment (alexeagle) [inline modules/@angular/compiler/test/test_component_builder_spec.ts]: some extra changes not required for lint?

File: modules/@angular/compiler/test/css_visitor_spec.ts (hunk 1 above)
Comment (vicb) [inline modules/@angular/compiler/test/css_visitor_spec.ts]: A couple of occurences, we could fix later

File: modules/@angular/core/test/linker/query_integration_spec.ts [hunk 3]
```diff
@@ -833,6 +831,7 @@ class NeedsViewChildren implements AfterViewInit {
 
 @Component({selector: 'needs-content-child', template: ''})
 class NeedsContentChild implements AfterContentInit, AfterContentChecked {
+  /** @internal */
```
Comment (alexeagle) [inline modules/@angular/core/test/linker/query_integration_spec.ts]: just remove the leading underscore instead?

File: modules/@angular/compiler/test/test_component_builder_spec.ts (hunk 2 above)
Comment (vicb) [inline modules/@angular/compiler/test/test_component_builder_spec.ts]: nope, diff is lost

**Pull request merged**